commands/build.go

```diff
@@ -643,6 +653,90 @@ func parsePrintFunc(str string) (*build.PrintFunc, error) {
 	return f, nil
 }
 
+func addGitProvenance(ctx context.Context, contextPath string, dockerfilePath string) ([]string, error) {
+	v, ok := os.LookupEnv("BUILDX_GIT_LABELS")
+	if !ok || contextPath == "" {
+		return nil, nil
+	}
+	labels := make([]string, 0)
+
+	// figure out in which directory the git command needs to run in
+	var wd string
+	if filepath.IsAbs(contextPath) {
+		wd = contextPath
+	} else {
+		cwd, _ := os.Getwd()
+		wd, _ = filepath.Abs(filepath.Join(cwd, contextPath))
+	}
+
+	// check if inside git working tree
+	cmd := exec.CommandContext(ctx, "git", "rev-parse", "--is-inside-work-tree")
+	cmd.Dir = wd
+	err := cmd.Run()
+	if err != nil {
+		logrus.Warnf("Unable to determine Git information")
+		return nil, nil
+	}
+
+	// obtain Git sha of current HEAD
+	cmd = exec.CommandContext(ctx, "git", "rev-parse", "HEAD")
+	cmd.Dir = wd
+	out, err := cmd.Output()
+	if err != nil {
+		return nil, errors.Errorf("error obtaining git head: %v", err)
```

Comment:
`errors.Wrap(err, "error obtaining.. `

Same for the rest of the errors

Reply:
Ah got it. I was just trying to use the same pattern that is used elsewhere in `build.go`. Every other location uses `errors.Errorf`.

Happy to change my code though.